# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: js/src/jit-test/tests/asm.js/bug1924062.js
# Commit: 3c25ee26a7ad
# Full Hash: 3c25ee26a7adab3e323b7f0409a7b4dca2f0f9f8
# Author: Jan de Mooij <jdemooij@mozilla.com>
# Date: 2024-10-15 04:16:11
# Description:
#   Bug 1924062 - Throw exception for asm.js module functions in disnative testing function. r=arai
#   
#   A function marked `isAsmJSNative` is either the outer module function or an actual
#   exported asm.js function. The module function can't be disassembled.
#   
# ==============================================================================

diff -r 3613fc672c50 -r 3c25ee26a7ad js/src/jit-test/tests/asm.js/bug1924062.js
--- /dev/null	Thu Jan 01 00:00:00 1970 +0000
+++ b/js/src/jit-test/tests/asm.js/bug1924062.js	Mon Oct 14 10:38:58 2024 +0000
@@ -0,0 +1,9 @@
+function a() {
+  "use asm"
+  function b() {}
+  return b;
+}
+// asm.js module.
+try { disnative(a); } catch (e) {}
+// asm.js exported function.
+try { disnative(a()); } catch (e) {}
